File: 383-ransom-note/ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] ch=new int[26];
        for(char i:magazine.toCharArray()){
            ch[i-'a']++;
        }for(char i:ransomNote.toCharArray()){
            if(ch[i-'a']==0){
                return false;
            }
                ch[i-'a']--;
            
        }
        return true;
        
    }
}

File: 383-ransom-note/ransom-note.c
bool canConstruct(char* r, char* m) {
    int c[26]={0};
    for(int i=0;i<strlen(m);i++){
        c[m[i]-'a']++;
    }
    for(int i=0;i<strlen(r);i++){
       // c[m[i]-'a']--;
        if(--c[r[i]-'a']<0){
            return false;
        }
        
    }
    return true;
}